src/fairseq2/checkpoint.py

```diff
@@ -534,6 +547,16 @@ def get_last_step_number(self, *, with_model: bool = False) -> Optional[int]:
 
         return None
 
+    @override
+    def get_model_checkpoint_path(
+        self, step_nr: Optional[int] = None
+    ) -> Optional[Path]:
+        if step_nr is None:
+            step_nr = self.get_last_step_number()
```

Comment:
A subtle bug, this should ideally be `self.get_last_step_number(with_model=True)`.